paasta_tools/cli/cmds/spark_run.py

```diff
@@ -111,6 +134,13 @@ def add_subparser(subparsers):
         help="Use the provided image to start the Spark driver and executors.",
     )
 
+    list_parser.add_argument(
+        "-e",
+        "--enable-k8s-autogen",
```

Comment:
The help section variable name and description should cater towards requirements of the tool users.
In this case, most of the paasta spark-run users are spark-batch implementors or core-ml (onpointers).

I think the name `enable-k8s-autogen` may be overloaded with keywords which they are either not familiar with or don't care. Example: autogen is for auto-generation of pod template and k8s for kubernetes.

Our mission is to hide the complexity of these underneath infra components as much as possible.
One way I tend to think is the functional aspect of the flag. 
In this case, I think the flag caters to `enable` or `disable` preferred launching of executors on same or different nodes (bin-packing vs round-robin).

1. Maybe something like, `--enable-bin-packing` or something else is more relevant name?

2.  It seems like by default we want to enable `bin-packing`.  Maybe in that case it should be `--disable-bin-packing`

Thoughts?

Reply:
An additional thing to check during exec time is that we generate this configuration only with `--cluster-manager=kubernetes`. This should also be mentioned in the `help` message as well, that with Mesos this flag has no effect.

Reply:
I agree with the naming being overloaded, however bin packing itself is already enabled by default and what we're doing is creating a tighter bin packing with pod affinity. Maybe something like `enable-compact-bin-packing`, I'd also like this to be defaulted to false so we can roll it out in the background.

Reply:
`enable-copact-bin-packing` sounds great. 

> ...I'd also like this to be defaulted to false so we can roll it out in the background.
Not sure I follow this part. Wouldn't we want it to enable by `default`, if we are rolling it out in the background?

During the migration, would every migration need to specify `--enable-compact-bin-packing`, if they need to keep the behavior as status-quo?

Reply:
I guess either way the scheduling behavior is probably a bit different from mesos, but the -e would have to be specified yeah, we can do a pr to add '-e' in all-repo-fixers if we decide to enable that for all jobs

Reply:
From functionality perspective, looks like the default value of `-e` doesn't matter. 

1. What is going to be the more preferred way of using this flag `-e` by default in `adhoc` or `deployment mode`?
2. Which is more closer behavior of the value of `-e` flag with status-quo? `True` or `False`.
3. What are the downsides of keeping `-e` flag `True` by default?

What it looks like to me so far is:

* For 1., it seems `-e` is going to be set to `True` for all batches. 
Why do we want to add the additional overhead for migration and all users in `adhoc` mode, to set `-e True`. 
It seems`users` would need to adapt to the new flags for paasta spark-run.

* For 2. It seems to `True`. Unless I am missed something here?

Reply:
I guess since we are adding `--cluster-manager kubernetes` on a case by case basis, we can add the `-e` flag in the meantime. The biggest downsides usually lies in the fact that the scheduler will need to compute the weight of each node and prioritise accordingly which can be an expensive when there are a lot of scheduling requests. I think as a feature it's safer to roll out in the background instead of making it the default.

Reply:
I take back what I said above.

I think in short-term it's ok to rollout default as `False`. 
The benefit of this is, the already live-launched batches are at `less risk`. 

On more thought, I agree when we come around `repo-fixers`, with `--cluster-manager kubernetes`, it is not a big-deal to `-e True`.

However, in long-term, we would definitely, plan to get rid of both the flags (--cluster-manager and -e) and set the preferred default values.

Reply:
I apologize for initially not thinking through from the perspective of `short-term` or `long-term` approach. 
Resolving this comment.